src/convertToObject.js

```diff
@@ -6,7 +6,23 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+
+  const splittedSource = sourceString.split(';');
+
+  for (let i = 0; i < splittedSource.length; i++) {
+    const rule = splittedSource[i].trim();
+
+    if (rule) {
+      const [key, value] = rule.split(':').map((part) => part.trim());
+
+      if (key && value !== undefined) {
+        result[key] = value;
+      }
+    }
+  }
```

Comment:
Using a 'for' loop is not recommended for this task. Instead, use array iteration methods such as 'forEach' after filtering out empty rules. This will make the code more concise and functional in style.